src/components/organisms/BannerAdmin.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useCallback, useRef } from "react";
+
+import { makeUpdateBanner } from "api/bannerAdmin";
+
+import { currentVenueSelector } from "utils/selectors";
+
+import { useSelector } from "hooks/useSelector";
+
+export const BannerAdmin: React.FC = () => {
+  const currentVenue = useSelector(currentVenueSelector);
+  const venueId = currentVenue?.id;
+  const existingBannerMessage = currentVenue?.bannerMessage ?? "";
+
+  const inputFieldRef = useRef<HTMLInputElement>(null);
```

Comment:
Is there a particular reason to use `ref` instead of `useState` ?

Reply:
We're not storing any state here, but we are getting a reference to the `<input>` element/DOM node itself:

- https://reactjs.org/docs/hooks-reference.html#useref
  - > You might be familiar with refs primarily as a way to access the DOM. If you pass a `ref` object to React with `<div ref={myRef} />`, React will set its `.current` property to the corresponding DOM node whenever that node changes.

Reply:
We're not storing any state here, but we are getting a reference to the `<input>` element/DOM node itself:

- https://reactjs.org/docs/hooks-reference.html#useref
  - > You might be familiar with refs primarily as a way to access the DOM. If you pass a `ref` object to React with `<div ref={myRef} />`, React will set its `.current` property to the corresponding DOM node whenever that node changes.
  
This is being used in the `useCallback`'d `saveBanner` function to reference `inputFieldRef.current.value`, so that the `useCallback` memo doesn't change on every update to the input field (because a ref always has a stable reference for comparisons, only it's `.current` changes reference)